src/screens/LunchScreen.js

```diff
@@ -0,0 +1,155 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useContext, useState } from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  RefreshControl,
+  StyleSheet,
+  TouchableOpacity
+} from 'react-native';
+
+import { colors, consts, normalize, texts } from '../config';
+import {
+  BoldText,
+  Button,
+  ConnectedImagesCarousel,
+  HeaderLeft,
+  Icon,
+  LunchSection,
+  RegularText,
+  SafeAreaViewFlex,
+  Wrapper,
+  WrapperRow,
+  WrapperWithOrientation
+} from '../components';
+import { arrowLeft, arrowRight } from '../icons';
+import { useMatomoTrackScreenView } from '../hooks';
+import { graphqlFetchPolicy } from '../helpers';
+import moment from 'moment';
+import { NetworkContext } from '../NetworkProvider';
+import { useQuery } from 'react-apollo';
+import { getQuery, QUERY_TYPES } from '../queries';
+
+const { MATOMO_TRACKING } = consts;
+
+export const LunchScreen = ({ navigation }) => {
+  const [poiId, setPoiId] = useState(navigation.getParam('poiId'));
+
+  const [date, setDate] = useState(moment());
+
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
+
+  const currentDate = moment(date).format('YYYY-MM-DD');
+
+  const variables = {
+    dateRange: [currentDate, currentDate]
+  };
+
+  const { data, loading, refetch } = useQuery(getQuery(QUERY_TYPES.LUNCHES), {
+    fetchPolicy,
+    variables
+  });
+
+  useMatomoTrackScreenView(MATOMO_TRACKING.SCREEN_VIEW.LUNCH);
+
+  const onPressNext = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).add(1, 'day'));
+  }, [setDate]);
+
+  const onPressPrevious = useCallback(() => {
+    setDate((oldDate) => moment(oldDate).subtract(1, 'day'));
+  }, [setDate]);
+
+  const renderItem = useCallback(
+    ({ item }) => <LunchSection lunchOfferData={item} navigation={navigation} />,
+    [navigation]
+  );
+
+  let lunchData = data?.[QUERY_TYPES.LUNCHES];
+
+  if (poiId) {
+    lunchData?.filter((item) => item?.pointOfInterest?.id === poiId);
+  }
+
+  const ListHeaderComponent = (
+    <>
+      <ConnectedImagesCarousel
+        navigation={navigation}
+        publicJsonFile="lunchCarousel"
+        refreshTimeKey="publicJsonFile-lunchCarousel"
+      />
```

Comment:
the carousel seems to update with ever date change forward/backwards. is that right? the images are not related to the lunch dates, right? could the carousel update with date change be avoided?

Reply:
i tried around for a little bit, but the only thing that comes to mind requires a bigger refactor where we move the date and list section to its own component with its own state and throw everything in a scrollview together. 
that would require similar refreshing logic as newly introduced on the home screen, but should work okay afterwards.
shall i do it?

Reply:
no